src/Symfony/Component/DependencyInjection/ContainerInterface.php

```diff
@@ -37,7 +37,9 @@ public function set(string $id, ?object $service): void;
      *
      * @param B $invalidBehavior
      *
-     * @psalm-return (B is self::EXCEPTION_ON_INVALID_REFERENCE|self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE ? object : object|null)
+     * @psalm-return ($id is class-string<A> ? (B is self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE|self::EXCEPTION_ON_INVALID_REFERENCE ? A : A|null) : (B is self::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE|self::EXCEPTION_ON_INVALID_REFERENCE ? object : object|null))
+     *
+     * @return ($id is class-string<A> ? (B is 0|1 ? A : A|null) : (B is 0|1 ? object : object|null))
```

Comment:
VS code does not understand the psalm code, but this one is almost okay.

![image](https://github.com/user-attachments/assets/186b64dd-1bde-4408-a008-9ee856641a94)

![image](https://github.com/user-attachments/assets/ecbec22e-19b6-43b4-900d-77fa7a1cbcb8)

Reply:
did you try with phpstan-return?